test/shared/src/main/scala/zio/test/TimeVariants.scala

```diff
@@ -33,16 +49,133 @@ trait TimeVariants {
+    for {
+      hour   <- Gen.int(0, 23)
+      minute <- Gen.int(0, 59)
+      second <- Gen.int(0, 59)
+      nanos  <- Gen.int(0, 999999999)
+    } yield LocalTime.of(hour, minute, second, nanos)
+
   /**
    * A generator of `java.time.LocalDateTime` values. Shrinks toward `LocalDateTime.MIN`.
    */
   final def anyLocalDateTime: Gen[Random, LocalDateTime] = localDateTime(LocalDateTime.MIN, LocalDateTime.MAX)
 
+  /**
+   * A generator of `java.time.Month` values. Shrinks toward `Month.JANUARY`.
+   */
+  final def anyMonth: Gen[Random, Month] =
+    Gen.elements(
+      Month.JANUARY,
+      Month.FEBRUARY,
+      Month.MARCH,
+      Month.APRIL,
+      Month.MAY,
+      Month.JUNE,
+      Month.JULY,
+      Month.AUGUST,
+      Month.SEPTEMBER,
+      Month.OCTOBER,
+      Month.NOVEMBER,
+      Month.DECEMBER
+    )
+
+  /**
+   * A generator of `java.time.MonthDay` values. Shrinks toward `MonthDay.of(Month.JANUARY, 1)`.
+   */
+  final def anyMonthDay: Gen[Random, MonthDay] =
+    for {
+      month <- Gen.int(1, 12).map(Month.of)
+      days  <- Gen.int(1, month.maxLength())
+    } yield MonthDay.of(month, days)
+
   /**
    * A generator of `java.time.OffsetDateTime` values. Shrinks toward `OffsetDateTime.MIN`.
    */
   final def anyOffsetDateTime: Gen[Random, OffsetDateTime] = offsetDateTime(OffsetDateTime.MIN, OffsetDateTime.MAX)
 
+  /**
+   * A generator of `java.time.OffsetTime` values. Shrinks torward `OffsetTime.MIN`.
+   */
+  final def anyOffsetTime: Gen[Random, OffsetTime] =
+    for {
+      time   <- anyLocalTime
+      offset <- anyZoneOffset
+    } yield OffsetTime.of(time, ZoneOffset.ofTotalSeconds(-offset.getTotalSeconds))
+
+  /**
+   * A generator of `java.time.Period` values. Shrinks toward `Period.ZERO`.
+   */
+  final def anyPeriod: Gen[Random, Period] =
+    for {
+      years  <- Gen.int(0, Int.MaxValue)
+      months <- Gen.int(0, Int.MaxValue)
+      days   <- Gen.int(0, Int.MaxValue)
+    } yield Period.of(years, months, days)
+
+  /**
+   * A generator of `java.time.Year` values. Shrinks toward `Year.of(Year.MIN_VALUE)`.
+   */
+  final def anyYear: Gen[Random, Year] =
+    Gen.int(Year.MIN_VALUE, Year.MAX_VALUE).map(Year.of)
+
+  /**
+   * A generator of `java.time.YearMonth` values. Shrinks toward `YearMonth.of(Year.MIN_VALUE, Month.JANUARY)`.
+   */
+  final def anyYearMonth: Gen[Random, YearMonth] =
+    for {
+      year  <- anyYear
+      month <- Gen.int(1, 12)
+    } yield YearMonth.of(year.getValue(), month)
+
+  /**
+   * A generator of `java.time.ZonedDateTime` values. Shrinks toward `ZoneDateTime.of(LocalDateTime.MIN, zoneId)`.
+   */
+  final def anyZonedDateTime: Gen[Random, ZonedDateTime] =
+    for {
+      dateTime <- anyLocalDateTime
+      zoneId   <- anyZoneId
+    } yield ZonedDateTime.of(dateTime, zoneId)
+
+  /**
+   * A generator of `java.time.ZoneId` values. Doesn't have any shrinking.
+   */
+  final def anyZoneId: Gen[Random, ZoneId] =
```

Comment:
It looks like we have separate versions specific implementations of this that seem to be identical and then this version. I think we should be able to just have one implementation of this that is like the version specific ones currently and just uses `import scala.collection.JavaConverters._` and `@silent("JavaConverters")`.

Reply:
Oh, didn't know about `@silent` annotation! Thank you Adam! 
Will do as you suggested)

Reply:
Oh my... I forgot to remove `anyZoneId` from `TimeVariants` when it was implemented using `ZStream` This looks somewhat ugly(

Reply:
@adamgfraser done!